src/types.ts

```diff
@@ -187,14 +188,7 @@ export type BaseAuth0APIClient = {
 }; // TODO: replace with a more accurate representation of the Auth0APIClient type
 
 export type Auth0APIClient = BaseAuth0APIClient & {
-  pool: {
-    addSingleTask: (arg0: { data: Object; generator: any }) => {
-      promise: () => Promise<ApiResponse>;
-    };
-    addEachTask: (arg0: { data: Object; generator: any }) => {
-      promise: () => Promise<Asset[][]>;
-    };
-  };
+  pool: PromisePoolExecutor;
```

Comment:
Adds more robust typing to the connection pool's `pool` property. All type modifications elsewhere in this PR are related to this change.